File: 0x05/BOJ10828.js
const readline = require('readline');
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let lineCnt = 0;
let n = 0;
let res = [];

class Stack {
    constructor() {
        this._arr = [];
    }
    push(data) {
        this._arr.push(data);
    }
    pop() {
        let res = this._arr.pop();
        if (res == undefined) return -1;
        else return Number(res);
    }
    top() {
        if (this._arr[this._arr.length - 1] == undefined) return -1;
        else return Number(this._arr[this._arr.length - 1]);
    }
    empty() {
        if (this._arr.length == 0) return 1;
        else return 0;
    }
    size() {
        return Number(stack._arr.length);
    }
}

let stack = new Stack;

rl.on('line', (line) => {

    if (lineCnt === 0) {
        n = Number(line);
        lineCnt++;
    }
    else {
        switch (line.split(' ')[0]) {
          case "push":
              stack.push(Number(line.split(' ')[1]));
              lineCnt++;
              break; 
          case "pop":
              res.push(stack.pop());
              lineCnt++;
              break;
          case "size":
              res.push(stack.size());
              lineCnt++;
              break;
          case "empty":
              res.push(stack.empty());
              lineCnt++;
              break;
          case "top":
              res.push(stack.top());
              lineCnt++;
              break;
                      
    }
    if (lineCnt === n + 1) {
      rl.close();
    }
    
}
}).on('close', () => {
    console.log(res.join('\n'));
    process.exit();
});

File: CPP/0x05/BOJ10828.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
	int n = 0;
	cin >> n;

	stack<int> S;

	while(n--) {
		string cmd;
		cin >> cmd;
		if (cmd == "push") {
			int x;
			cin >> x;
			S.push(x);
		}
		else if (cmd == "pop") {
			if(S.empty()) {
				cout << -1 << endl;
			}
			else {
				cout << S.top() << endl;
				S.pop();
			}
		}
		else if (cmd == "size") {
			cout << S.size() << endl;
		}
		else if (cmd == "empty") {
			if (S.empty()) cout << 1 << endl;
			else cout << 0 << endl;
		}
		else {
			if (S.empty()) cout << -1 << endl;
			else cout << S.top() << endl;
		}
	}
}
